src/main/scala/beam/utils/scenario/UrbanSimScenarioLoader.scala

```diff
@@ -29,30 +31,40 @@ class UrbanSimScenarioLoader(
   val geo: GeoUtils
 ) extends LazyLogging {
 
+  implicit val ex: ExecutionContext = scala.concurrent.ExecutionContext.Implicits.global
+
   val population: Population = scenario.getPopulation
 
   val availableModes: String = BeamMode.allModes.map(_.value).mkString(",")
 
   def loadScenario(): Scenario = {
     clear()
 
-    val plans = scenarioSource.getPlans
-    logger.info(s"Read ${plans.size} plans")
-
-    val personsWithPlans = {
+    val plansF = Future {
+      val plans = scenarioSource.getPlans
+      logger.info(s"Read ${plans.size} plans")
+      plans
+    }
+    val personsF = Future {
       val persons: Iterable[PersonInfo] = scenarioSource.getPersons
       logger.info(s"Read ${persons.size} persons")
-      getPersonsWithPlan(persons, plans)
+      persons
+    }
+    val householdsF = Future {
+      val households = scenarioSource.getHousehold
+      logger.info(s"Read ${households.size} households")
+      households
     }
+    val plans = Await.result(plansF, 500.seconds)
+    val persons = Await.result(personsF, 500.seconds)
+
+    val personsWithPlans = getPersonsWithPlan(persons, plans)
```

Comment:
Can we keep the flow going instead of awaiting? Await is a smell, so if we can push the futures all the way until they are needed then that's probably better:

i.e.

```
val personsWithPlansF = for{
  plans <- plansF
  persons <- personsF
} yield getPersonsWithPlan(persons, plans)
```

And so on until the future is forced.

Reply:
Agree than almost always blocking for the result is a bad idea, but in this particular case we can't do anything _better_ because the code is not async all the way. To be able to run the simulation we need to have properly populated scenario.  Also, `val population: Population = scenario.getPopulation` is Java class and it is blackbox for me, so I can't do side-effects on this class in the thread other than `main` because I'm not sure will be the changes properly visible in `main` thread (Future executes in another thread).